hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -574,7 +574,20 @@ private boolean shouldRetryRead(IOException cause) throws IOException {
     } catch (Exception e) {
       throw new IOException(e);
     }
-    return retryAction.action == RetryPolicy.RetryAction.RetryDecision.RETRY;
+    if (retryAction.action == RetryPolicy.RetryAction.RetryDecision.RETRY) {
```

Comment:
Just a thought: It might be nice if we can have a general RetryAction handler so that we don't need to duplicate the sleeping logic (similar codes is in `AbstractDataStreamOutput#shouldRetry`, `GrpcOmTranport#shouldRetry`, and `KeyOutputStream#handleRetry`)

Reply:
Yes sounds good may be we can refactor in other Jira.